Migrations: Differ doesn't use configured ValueComparer for DefaultValue

## File a bug

### Include your code

[Project that reproduce bug](https://github.com/Razorik/not-removing-migrations-example)

#### Steps to reproduce

1. Start postgres with command `docker-compose up -d` in solution root directory
2. Uncomment lines 11-13 and 21-25 in ExampleContext.cs file (skip this step if you want to make sure that migrations can be created and removed)
3. Go to App directory
4. Run command `dotnet ef migrations add FirstMigration` (or any another migration name)
5. Run command `dotnet ef migrations remove`

### Include verbose output

#### Output with commented lines 11-13 and 21-25
```
melkozerov@MacBook-Pro-Ivan-3 App % dotnet ef migrations add FirstMigration --verbose
Using project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Using startup project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpd1NR3y.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpacv7mU.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Build started...
dotnet build /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj /verbosity:quiet /nologo

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.34
Build succeeded.
dotnet exec --depsfile /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.deps.json --additionalprobingpath /Users/melkozerov/.nuget/packages --runtimeconfig /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.runtimeconfig.json /Users/melkozerov/.dotnet/tools/.store/dotnet-ef/6.0.10/dotnet-ef/6.0.10/tools/net6.0/any/tools/netcoreapp2.0/any/ef.dll migrations add FirstMigration --assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --startup-assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --startup-project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --project-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ --root-namespace App --language C# --framework net6.0 --nullable --working-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App --verbose
Using assembly 'App'.
Using startup assembly 'App'.
Using application base '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0'.
Using working directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App'.
Using root namespace 'App'.
Using project directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/'.
Remaining arguments: .
Finding DbContext classes...
Finding IDesignTimeDbContextFactory implementations...
Found IDesignTimeDbContextFactory implementation 'ContextFactory'.
Found DbContext 'ExampleContext'.
Finding application service provider in assembly 'App'...
Finding Microsoft.Extensions.Hosting service provider...
Using environment 'Development'.
Using application service provider from Microsoft.Extensions.Hosting.
Finding DbContext classes in the project...
Using DbContext factory 'ContextFactory'.
Using context 'ExampleContext'.
Finding design-time services referenced by assembly 'App'...
Finding design-time services referenced by assembly 'App'...
No referenced design-time services were found.
Finding design-time services for provider 'Npgsql.EntityFrameworkCore.PostgreSQL'...
Using design-time services from provider 'Npgsql.EntityFrameworkCore.PostgreSQL'.
Finding IDesignTimeServices implementations in assembly 'App'...
No design-time services were found.
DetectChanges starting for 'ExampleContext'.
DetectChanges completed for 'ExampleContext'.
Writing migration to '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/Migrations/20221025080735_FirstMigration.cs'.
Writing model snapshot to '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/Migrations/ExampleContextModelSnapshot.cs'.
'ExampleContext' disposed.
Done. To undo this action, use 'ef migrations remove'


melkozerov@MacBook-Pro-Ivan-3 App % dotnet ef migrations remove --verbose
Using project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Using startup project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpnsC7nw.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpbBKCBF.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Build started...
dotnet build /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj /verbosity:quiet /nologo

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.89
Build succeeded.
dotnet exec --depsfile /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.deps.json --additionalprobingpath /Users/melkozerov/.nuget/packages --runtimeconfig /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.runtimeconfig.json /Users/melkozerov/.dotnet/tools/.store/dotnet-ef/6.0.10/dotnet-ef/6.0.10/tools/net6.0/any/tools/netcoreapp2.0/any/ef.dll migrations remove --assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --startup-assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --startup-project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --project-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ --root-namespace App --language C# --framework net6.0 --nullable --working-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App --verbose
Using assembly 'App'.
Using startup assembly 'App'.
Using application base '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0'.
Using working directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App'.
Using root namespace 'App'.
Using project directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/'.
Remaining arguments: .
Finding DbContext classes...
Finding IDesignTimeDbContextFactory implementations...
Found IDesignTimeDbContextFactory implementation 'ContextFactory'.
Found DbContext 'ExampleContext'.
Finding application service provider in assembly 'App'...
Finding Microsoft.Extensions.Hosting service provider...
Using environment 'Development'.
Using application service provider from Microsoft.Extensions.Hosting.
Finding DbContext classes in the project...
Using DbContext factory 'ContextFactory'.
Using context 'ExampleContext'.
Finding design-time services referenced by assembly 'App'...
Finding design-time services referenced by assembly 'App'...
No referenced design-time services were found.
Finding design-time services for provider 'Npgsql.EntityFrameworkCore.PostgreSQL'...
Using design-time services from provider 'Npgsql.EntityFrameworkCore.PostgreSQL'.
Finding IDesignTimeServices implementations in assembly 'App'...
No design-time services were found.
DetectChanges starting for 'ExampleContext'.
DetectChanges completed for 'ExampleContext'.
DetectChanges starting for 'ExampleContext'.
DetectChanges completed for 'ExampleContext'.
Opening connection to database 'example' on server ''.
Opened connection to database 'example' on server 'tcp://localhost:5432'.
Closing connection to database 'example' on server 'tcp://localhost:5432'.
Closed connection to database 'example' on server ''.
Creating DbCommand for 'ExecuteScalar'.
Created DbCommand for 'ExecuteScalar' (2ms).
Opening connection to database 'example' on server ''.
Opened connection to database 'example' on server 'tcp://localhost:5432'.
Executing DbCommand [Parameters=[], CommandType='Text', CommandTimeout='30']
SELECT EXISTS (SELECT 1 FROM pg_catalog.pg_class c JOIN pg_catalog.pg_namespace n ON n.oid=c.relnamespace WHERE c.relname='__EFMigrationsHistory');
Executed DbCommand (88ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
SELECT EXISTS (SELECT 1 FROM pg_catalog.pg_class c JOIN pg_catalog.pg_namespace n ON n.oid=c.relnamespace WHERE c.relname='__EFMigrationsHistory');
Closing connection to database 'example' on server 'tcp://localhost:5432'.
Closed connection to database 'example' on server ''.
Removing migration '20221025080735_FirstMigration'.
Removing model snapshot.
Done.
'ExampleContext' disposed.
```

#### Output with uncommented lines 11-13 and 21-25
```
melkozerov@MacBook-Pro-Ivan-3 App % dotnet ef migrations add FirstMigration --verbose
Using project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Using startup project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpAJvTzH.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpb9f0J3.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Build started...
dotnet build /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj /verbosity:quiet /nologo
/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,84): warning CS8604: Possible null reference argument for parameter 'first' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]
/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,101): warning CS8604: Possible null reference argument for parameter 'second' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]

Build succeeded.

/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,84): warning CS8604: Possible null reference argument for parameter 'first' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]
/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,101): warning CS8604: Possible null reference argument for parameter 'second' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.83
Build succeeded.
dotnet exec --depsfile /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.deps.json --additionalprobingpath /Users/melkozerov/.nuget/packages --runtimeconfig /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.runtimeconfig.json /Users/melkozerov/.dotnet/tools/.store/dotnet-ef/6.0.10/dotnet-ef/6.0.10/tools/net6.0/any/tools/netcoreapp2.0/any/ef.dll migrations add FirstMigration --assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --startup-assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --startup-project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --project-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ --root-namespace App --language C# --framework net6.0 --nullable --working-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App --verbose
Using assembly 'App'.
Using startup assembly 'App'.
Using application base '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0'.
Using working directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App'.
Using root namespace 'App'.
Using project directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/'.
Remaining arguments: .
Finding DbContext classes...
Finding IDesignTimeDbContextFactory implementations...
Found IDesignTimeDbContextFactory implementation 'ContextFactory'.
Found DbContext 'ExampleContext'.
Finding application service provider in assembly 'App'...
Finding Microsoft.Extensions.Hosting service provider...
Using environment 'Development'.
Using application service provider from Microsoft.Extensions.Hosting.
Finding DbContext classes in the project...
Using DbContext factory 'ContextFactory'.
Using context 'ExampleContext'.
Finding design-time services referenced by assembly 'App'...
Finding design-time services referenced by assembly 'App'...
No referenced design-time services were found.
Finding design-time services for provider 'Npgsql.EntityFrameworkCore.PostgreSQL'...
Using design-time services from provider 'Npgsql.EntityFrameworkCore.PostgreSQL'.
Finding IDesignTimeServices implementations in assembly 'App'...
No design-time services were found.
DetectChanges starting for 'ExampleContext'.
DetectChanges completed for 'ExampleContext'.
Writing migration to '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/Migrations/20221025080845_FirstMigration.cs'.
Writing model snapshot to '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/Migrations/ExampleContextModelSnapshot.cs'.
'ExampleContext' disposed.
Done. To undo this action, use 'ef migrations remove'


melkozerov@MacBook-Pro-Ivan-3 App % dotnet ef migrations remove --verbose            
Using project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Using startup project '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj'.
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpd4AxqE.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Writing '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/obj/App.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/hk/tknx4grs5tl622c8myjb2hv40000gn/T/tmpH5m2Dm.tmp /verbosity:quiet /nologo /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj
Build started...
dotnet build /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj /verbosity:quiet /nologo
/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,84): warning CS8604: Possible null reference argument for parameter 'first' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]
/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,101): warning CS8604: Possible null reference argument for parameter 'second' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]

Build succeeded.

/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,84): warning CS8604: Possible null reference argument for parameter 'first' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]
/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ExampleContext.cs(11,101): warning CS8604: Possible null reference argument for parameter 'second' in 'bool Enumerable.SequenceEqual<Flag>(IEnumerable<Flag> first, IEnumerable<Flag> second)'. [/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj]
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.74
Build succeeded.
dotnet exec --depsfile /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.deps.json --additionalprobingpath /Users/melkozerov/.nuget/packages --runtimeconfig /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.runtimeconfig.json /Users/melkozerov/.dotnet/tools/.store/dotnet-ef/6.0.10/dotnet-ef/6.0.10/tools/net6.0/any/tools/netcoreapp2.0/any/ef.dll migrations remove --assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --startup-assembly /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0/App.dll --startup-project /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/App.csproj --project-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/ --root-namespace App --language C# --framework net6.0 --nullable --working-dir /Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App --verbose
Using assembly 'App'.
Using startup assembly 'App'.
Using application base '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/bin/Debug/net6.0'.
Using working directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App'.
Using root namespace 'App'.
Using project directory '/Users/melkozerov/pet-repositories/NotRemovingMigrationsExample/App/'.
Remaining arguments: .
Finding DbContext classes...
Finding IDesignTimeDbContextFactory implementations...
Found IDesignTimeDbContextFactory implementation 'ContextFactory'.
Found DbContext 'ExampleContext'.
Finding application service provider in assembly 'App'...
Finding Microsoft.Extensions.Hosting service provider...
Using environment 'Development'.
Using application service provider from Microsoft.Extensions.Hosting.
Finding DbContext classes in the project...
Using DbContext factory 'ContextFactory'.
Using context 'ExampleContext'.
Finding design-time services referenced by assembly 'App'...
Finding design-time services referenced by assembly 'App'...
No referenced design-time services were found.
Finding design-time services for provider 'Npgsql.EntityFrameworkCore.PostgreSQL'...
Using design-time services from provider 'Npgsql.EntityFrameworkCore.PostgreSQL'.
Finding IDesignTimeServices implementations in assembly 'App'...
No design-time services were found.
The model snapshot and the backing model of the last migration are different. Continuing under the assumption that the last migration was deleted manually.
Reverting the model snapshot.
Done.
'ExampleContext' disposed.
```

### Include provider and version information

EF Core version: 6.0.10
dotnet-ef version: 6.0.10
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL 6.0.7
Target framework: .NET 6.0
Operating system: macOS 11.6 (20G165)
IDE: JetBrains Rider 2022.1.1, Build #RD-221.5591.20, built on April 28, 2022

--- a/src/EFCore.Relational/Migrations/Internal/MigrationsModelDiffer.cs
+++ b/src/EFCore.Relational/Migrations/Internal/MigrationsModelDiffer.cs
@@ -1040,7 +1040,8 @@ public class MigrationsModelDiffer : IMigrationsModelDiffer
             || source.DefaultValueSql != target.DefaultValueSql
             || source.ComputedColumnSql != target.ComputedColumnSql
             || source.IsStored != target.IsStored
-            || !Equals(sourceDefault, targetDefault)
+            || sourceDefault?.GetType() != targetDefault?.GetType()
+            || (sourceDefault != DBNull.Value && !target.ProviderValueComparer.Equals(sourceDefault, targetDefault))
             || source.Comment != target.Comment
             || source.Collation != target.Collation
             || source.Order != target.Order

--- a/test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs
+++ b/test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs
@@ -2960,6 +2960,27 @@ public class MigrationsModelDifferTest : MigrationsModelDifferTestBase
             });
 
     [ConditionalFact]
+    public void No_alter_column_default_when_references_not_equal()
+        => Execute(
+            source => source.Entity(
+                "Snake",
+                x =>
+                {
+                    x.Property<int>("Id");
+                    x.Property<byte[]>("Bytes")
+                        .HasDefaultValue(new byte[] { 0 });
+                }),
+            target => target.Entity(
+                "Snake",
+                x =>
+                {
+                    x.Property<int>("Id");
+                    x.Property<byte[]>("Bytes")
+                        .HasDefaultValue(new byte[] { 0 });
+                }),
+            Assert.Empty);
+
+    [ConditionalFact]
     public void Alter_column_default_expression()
         => Execute(
             source => source.Entity(

